Add example and notes and warnings sections (#808)
arduino/reference-en: Language/Functions/Communication/Serial/readString.adoc

=== Returns
A `String` read from the serial buffer

--
// OVERVIEW SECTION ENDS

// HOW TO USE SECTION STARTS
[#howtouse]
--

[float]
=== Example Code
Demonstrate Serial.readString()

[source,arduino]
----
void setup() {
  Serial.begin(9600);
}

void loop() {
  Serial.println("Enter data:");

>   while(Serial.available()==0); //wait for data available

Reviewer: ```suggestion
  while (Serial.available() == 0) {}     //wait for data available
```

### Pad operators

This is the standard Arduino code formatting style, as implemented by the default configuration of the Arduino IDE's "**Auto Format**" feature

### Adjust `while` block style

I replaced the semicolon with braces in `while` block.

I think it is best practices to always wrap flow control blocks in braces, since omitting them makes the code less clear and is frequently the cause of confusing bugs.

I also think the formatting looks nicer with this approach compared to how it looks after the Arduino IDE 2.x "**Auto Format**" (which is a bit more comprehensive than Arduino IDE 1.x):

```cpp
  while (Serial.available() == 0)
    ;                                    //wait for data available
```

>   String teststr = Serial.readString(); //read until timeout
>   teststr.trim(); // remove any \r \n whitespace at the end of the String
>   if(teststr=="red") {

Reviewer: ```suggestion
  String teststr = Serial.readString();  //read until timeout
  teststr.trim();                        // remove any \r \n whitespace at the end of the String
  if (teststr == "red") {
```

Bring code formatting into compliance with the standard Arduino code formatting style, as implemented by the default configuration of the Arduino IDE's "Auto Format" feature.

    Serial.println("A primary color");
  } else {
    Serial.println("Something else");
  }
}
----
[%hardbreaks]

[float]
=== Notes and Warnings

> The function does not terminate early because of end of line characters and the returned String may contain carriage return and/or line feed characters if they were received.

Reviewer: ```suggestion
The function does not terminate early if the data contains end of line characters. The returned `String` may contain carriage return and/or line feed characters if they were received.
```

Minor rewording and formatting for readability.
